src/map/mob.cpp

```diff
@@ -4073,10 +4073,19 @@ void mobskill_delay(mob_data& md, t_tick tick)
 		// If the skill cannot be found anymore because the monster's state has changed no delay will be applied
 		int32 delay = 0;
 		for (int32 i = 0; i < ms.size(); i++) {
-			if (ms[i]->state == md.state.skillstate && ms[i]->skill_id == ms[md.skill_idx]->skill_id) {
+			if (ms[i]->skill_id == ms[md.skill_idx]->skill_id) {
```

Comment:
I wonder, why don't we use foreach here?

Reply:
Yeah I thought the same.
Also it would require less indentation if we would invert the logic and skip entries.
But I did not want to suggest to change this in this PR.

Reply:
It would be purely stylistic and no functional change.

Reply:
Could still break something!

Reply:
Decided to finish this one without refactor. Mainly because I can't think of a good way to invert it, because all conditions would need to fail to continue. Feel free to make a follow-up PR!